core/src/test/java/org/jboss/sbomer/core/test/unit/GenericPurlWrapperUtilTest.java

```diff
@@ -0,0 +1,214 @@
+package org.jboss.sbomer.core.test.unit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.cyclonedx.model.component.evidence.Identity;
+import org.cyclonedx.model.component.evidence.Method;
+import org.cyclonedx.model.component.evidence.Method.Technique;
+import org.jboss.sbomer.core.features.sbom.utils.GenericPurlWrapperUtil;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.packageurl.MalformedPackageURLException;
+import com.github.packageurl.PackageURL;
+
+class GenericPurlWrapperUtilTest {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private static class Diff {
```

Comment:
You could use Lombok `@Getter` here although it also looks like nothing is accessing the fields.